napari/_qt/qt_main_window.py

```diff
@@ -771,20 +788,36 @@ def _add_menus(self):
         self._main_menu_shortcut.activated.connect(
             self._toggle_menubar_visible
         )
-
-        self.file_menu = menus.FileMenu(self)
+        # file menu
+        self.file_menu = build_qmodel_menu(
+            MenuId.MENUBAR_FILE, title=trans._('&File'), parent=self._qt_window
+        )
+        self._setup_npe1_samples_menu()
+        self.file_menu.aboutToShow.connect(
+            lambda: self._update_menu_state('file_menu')
+        )
         self.main_menu.addMenu(self.file_menu)
+        # view menu
         self.view_menu = build_qmodel_menu(
             MenuId.MENUBAR_VIEW, title=trans._('&View'), parent=self._qt_window
         )
+        self.view_menu.aboutToShow.connect(
+            lambda: self._update_menu_state('view_menu')
+        )
         self.main_menu.addMenu(self.view_menu)
+        # plugin menu
         self.plugins_menu = menus.PluginsMenu(self)
         self.main_menu.addMenu(self.plugins_menu)
+        # window menu
         self.window_menu = menus.WindowMenu(self)
         self.main_menu.addMenu(self.window_menu)
+        # help menu
         self.help_menu = build_qmodel_menu(
             MenuId.MENUBAR_HELP, title=trans._('&Help'), parent=self._qt_window
         )
+        self.help_menu.aboutToShow.connect(
```

Comment:
Remind me again why this function is needed here for the `help_menu` and `view_menu` but wasn't needed in the previous PRs?

Reply:
Second bullet point here: https://github.com/napari/napari/pull/4865#issuecomment-1635711551

They are not needed because no current actions in those menus use context, but added for posterity in case one is added in future, esp once we allow menu contributions from plugins.